lighthouse-core/scripts/pptr-run-devtools.js

```diff
@@ -152,122 +183,147 @@ const startLighthouse = `
+  const browser = await puppeteer.launch({
+    executablePath: getChromePath(),
+    args: chromeFlags,
+    devtools: true,
   });
 
-  if (config) {
-    // Must attach to the Lighthouse worker target and override the `self.createConfig`
-    // function, allowing us to use any config we want.
-    session.send('Target.setAutoAttach', {
-      autoAttach: true, flatten: true, waitForDebuggerOnStart: false,
-    });
-    session.once('Target.attachedToTarget', async (event) => {
-      if (event.targetInfo.type !== 'worker') throw new Error('expected lighthouse worker');
-
-      const targets = await browser.targets();
-      const workerTarget = targets.find(t => t._targetId === event.targetInfo.targetId);
-      if (!workerTarget) throw new Error('No lighthouse worker target found.');
-
-      const workerSession = await workerTarget.createCDPSession();
-      await Promise.all([
-        workerSession.send('Runtime.enable'),
-        workerSession.send('Debugger.enable'),
-        new Promise(resolve => {
-          workerSession.once('Debugger.scriptParsed', resolve);
-        }),
-      ]);
-      await workerSession.send('Debugger.pause');
-      await workerSession.send('Runtime.evaluate', {
-        expression: `self.createConfig = () => (${JSON.stringify(config)});`,
-      });
-      await workerSession.send('Debugger.resume');
-    });
-  }
+  try {
+    if ((await browser.version()).startsWith('Headless')) {
+      throw new Error('You cannot use headless');
+    }
 
-  /** @type {Omit<LH.Puppeteer.Protocol.Runtime.EvaluateResponse, 'result'>|undefined} */
-  let startLHResponse;
-  while (!startLHResponse || startLHResponse.exceptionDetails) {
-    if (startLHResponse) await new Promise(resolve => setTimeout(resolve, 1000));
-    startLHResponse = await session.send('Runtime.evaluate', {
-      expression: startLighthouse,
-      awaitPromise: true,
-    }).catch(err => ({exceptionDetails: err}));
-  }
+    const page = (await browser.pages())[0];
 
-  /** @type {LH.Puppeteer.Protocol.Runtime.EvaluateResponse} */
-  const lhStartedResponse = await session.send('Runtime.evaluate', {
-    expression: sniffLighthouseStarted,
-    awaitPromise: true,
-    returnByValue: true,
-  }).catch(err => err);
-  // Verify the first parameter to `startLighthouse`, which should be a url.
-  // In M100 the LHR is returned on `collectLighthouseResults` which has just 1 options parameter containing `inspectedUrl`.
-  // Don't try to check the exact value (because of redirects and such), just
-  // make sure it exists.
-  if (
-    !isValidUrl(lhStartedResponse.result.value) &&
-    !isValidUrl(lhStartedResponse.result.value.inspectedURL)
-  ) {
-    throw new Error(`Lighthouse did not start correctly. Got unexpected value for url: ${
-      JSON.stringify(lhStartedResponse.result.value)}`);
-  }
+    const inspectorTarget = await browser.waitForTarget(t => t.url().includes('devtools'));
```

Comment:
```js
inspectorTarget._targetInfo.type = 'page';
```

add that and then `inspectorTarget.page()` _should_ work fine.  i didn't test any APIs but ... the devtools UI is a page as far as the browser is concerned, so i thinkkkkk this'll be fine.


----

after figuring this out i saw Tony from Edge devtools team figured out the same hack: https://github.com/puppeteer/puppeteer/issues/4247


also in the current cdt e2e tests they create a new tab that inspects another tab: https://source.chromium.org/chromium/chromium/src/+/main:third_party/devtools-frontend/src/test/conductor/frontend_tab.ts;l=50-73;drc=db0dd9428654edf2c71c2979ce9b4ee8d0191eff
That approach would also work, I guess. But it relies on the weird hosted mode approach too much for my tastes.